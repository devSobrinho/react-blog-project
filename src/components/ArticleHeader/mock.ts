import { ArticleHeaderProps } from '.';
import mockArticleMeta from '../ArticleMeta/mock';
import mockHeader from '../Header/mock';

const header = {
  ...mockHeader,
  showText: false,
};

export default {
  header: header,
  title: 'Ipsum movet liberavisse an mea, diceret vituperata',
  excerpt:
    'Abhorreant reformidans eum an, per equidem repudiandae no, ex cum elitr argumentum persequeris.',
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  articleMeta: mockArticleMeta,
} as ArticleHeaderProps;
