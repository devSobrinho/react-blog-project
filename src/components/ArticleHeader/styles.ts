import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};
    padding-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    & ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      /* color: ${theme.colors.darkText}; */
      font-size: ${theme.font.sizes.huge};
    }

    @media${theme.media.lteMedium} {
      & ${HeadingStyles} {
        font-size: ${theme.font.sizes.xlarge};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: 2.4rem;
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
