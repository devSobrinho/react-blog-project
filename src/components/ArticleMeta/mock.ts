import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'daniel-sobrinho',
    displayName: 'Daniel Sobrinho',
  },
  categories: [
    {
      id: 'nnfgr3n434232',
      displayName: 'React',
      slug: 'react',
    },
    {
      id: 'a33d32d32fafa',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
