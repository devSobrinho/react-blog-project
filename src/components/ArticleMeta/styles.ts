import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    p {
      font-size: ${theme.font.sizes.small};
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    @media${theme.media.lteMedium} {
      p {
        font-size: ${theme.font.sizes.small};
      }
    }
  `}
`;
