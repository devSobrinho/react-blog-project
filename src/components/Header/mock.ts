import { HeaderProps } from '.';

export default {
  blogName: 'Daniel Sobrinho',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  showText: true,
  logo: {
    url: 'assets/images/dev-sobrinho.jpeg',
    id: 'qqqq11w1',
    alternativeText: 'textoalte',
  },
} as HeaderProps;
