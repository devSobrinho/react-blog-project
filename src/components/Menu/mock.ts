import { MenuProps } from '.';

export default {
  logoLink: {
    text: 'LogoLink',
    srcImg: 'assets/images/dev-sobrinho.jpeg',
    link: '/',
  },
  links: [
    { children: 'MenuLink1', link: '#home1', id: 'link-1', newTab: false },
    { children: 'MenuLink2aadsd', link: '#home2', id: 'link-2', newTab: false },
    { children: 'MenuLink3', link: '#home3', id: 'link-3', newTab: false },
    { children: 'MenuLink4asd', link: '#home4', id: 'link-4', newTab: false },
    {
      children: 'MenuLink5ssssssssssssssssssssssaaaa',
      link: '#home5',
      id: 'link-5',
      newTab: false,
    },
    { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLink6', link: '#home6', id: 'link-6', newTab: false },
    // { children: 'MenuLinkaaaaa', link: '#home6', id: 'link-6', newTab: false },
  ],
} as MenuProps;
