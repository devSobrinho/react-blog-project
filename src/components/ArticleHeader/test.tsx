import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleHeader, ArticleHeaderProps } from '.';
import mock from './mock';
import { formatDate } from '../../utils/format-data';

const props: ArticleHeaderProps = mock;

describe('<ArticleHeader />', () => {
  it('should render heading, excerpt, cover img and meta', () => {
    renderTheme(<ArticleHeader {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByText(props.excerpt)).toBeInTheDocument();
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.cover,
    );
    expect(
      screen.getByText(formatDate(props.articleMeta.createdAt)),
    ).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleHeader {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
