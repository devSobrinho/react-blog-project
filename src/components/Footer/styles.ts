import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
