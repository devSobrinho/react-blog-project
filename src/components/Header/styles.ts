import styled, { css } from 'styled-components';
import { Title as HeaddingStyles } from '../Heading/styles';

export const Content = styled.div`
  ${({ theme }) => css`
    max-width: 48rem;
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      margin-left: 0;

      & > h2 {
        text-align: center;
      }

  `}
`;
export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    padding-bottom: ${theme.spacings.xhuge};
    margin: 0 auto;

    & ${HeaddingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeaddingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }
    }
  `}
`;
