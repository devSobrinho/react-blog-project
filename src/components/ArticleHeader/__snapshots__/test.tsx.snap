// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader /> should match snapshot 1`] = `
.c3 {
  color: #000000;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c5 {
  color: #000000;
  font-size: 6.4rem;
  text-transform: none;
}

.c0 {
  margin: 0 auto;
  max-width: 96rem;
  padding-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 .c2 {
  margin: 0;
  margin-bottom: 2.4rem;
  font-size: 5.6rem;
}

.c6 {
  font-size: 2.4rem;
  margin: 2.4rem 0;
}

.c7 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

.c8 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c8 .categories span::after {
  content: ', ';
}

.c8 .categories span:last-child::after {
  content: '';
}

.c8 p {
  font-size: 1.6rem;
}

.c8 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c4 > img {
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  padding-bottom: 6.4rem;
  margin: 0 auto;
}

.c1 .c2 {
  margin: 0 0 calc(1.6rem - 1rem);
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c0 .c2 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c8 p {
    font-size: 1.6rem;
  }
}

<header
  class="c0"
>
  <header
    class="c1"
  >
    <h1
      class="c2 c3"
    >
      <a
        class="c4"
        href="/"
        target="_self"
      >
        <img
          alt="Daniel Sobrinho - Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud."
          src="assets/images/dev-sobrinho.jpeg"
        />
      </a>
    </h1>
  </header>
  <h1
    class="c2 c5"
  >
    Ipsum movet liberavisse an mea, diceret vituperata
  </h1>
  <p
    class="c6"
  >
    Abhorreant reformidans eum an, per equidem repudiandae no, ex cum elitr argumentum persequeris.
  </p>
  <img
    alt="Ipsum movet liberavisse an mea, diceret vituperata"
    class="c7"
    src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg"
  />
  <div
    class="c8"
  >
    <p>
      <span />
      <a
        href="/author/daniel-sobrinho"
      >
        Daniel Sobrinho
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2021-03-02T15:16:09.246+00:00"
      >
        2 de mar. de 2021
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/react"
          >
            React
          </a>
        </span>
        <span>
          <a
            href="/category/javascript"
          >
            JS
          </a>
        </span>
      </span>
    </p>
  </div>
</header>
`;
